todo-list-backend/src/index.ts

```diff
@@ -1,33 +1,208 @@
-import express, { Request, Response } from 'express';
+import express, { type Request, type Response } from 'express';
+import { readFileSync, writeFileSync } from 'fs'
+import type { TodoItem } from "./types";
+import { saveTodoList, readTodoList } from './fileIO';
+import { error } from 'console';
+import { json } from 'stream/consumers';
+import cors from 'cors';
 
 const app = express();
 const port = 3000;
 
-// State (count the number of requests)
+app.use(cors({
+  "methods": "GET,HEAD,PUT,PATCH,POST,DELETE,OPTIONS"
+}));
+
+// Just putting these here for debugging purposes
 let count = 0;
 
-// When someone makes a request to http://localhost:3000/, run this code, and give this reply
-app.get('/', (req: Request, res: Response) => {
+let todoItems: TodoItem[] = [];
+
+app.get("/", (req: Request, res: Response) => {
+    console.log("pinging")
     res
-    .setHeader('Access-Control-Allow-Origin', '*')
-    .status(200)
-    .send('Hello, TypeScript Express!');
+        .status(200)
+        .send("ping recieved");
 });
 
-app.get("/ping",  (req: Request, res: Response) => {
+// Just putting these here for debugging purposes
+app.get("/ping", (req: Request, res: Response) => {
     count++;
-    
+
     let responseContent = {
         message: "pong",
         count: count,
     };
-    
+
+    res
+        .status(200)
+        .json(responseContent);
+});
+
+app.get("/refresh-todos", (req: Request, res: Response) => {
+    console.log("refreshing todo list")
+    todoItems = readTodoList()
+    res
+        .status(200)
+        .send(JSON.stringify(todoItems))
+        ;
+});
```

Comment:
It's awesome that you've added functionality for saving/reading your to-do list from a file! However, I'm a little confused why the frontend would need to ask the backend to reload from the file. I'm nitpicking here, but in a real app it could be confusing to have the to-dos only read from data storage when the user clicks a button! Maybe the `readTodoList()` function could just be called once during app startup, to load in any existing todos?